#### Hypothesis 3: There has been an increase in discount provided across products, contributing to lower Median Order Value and inconsistent revenue growth.

In [1]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt

In [8]:
# loading the data
data = pd.read_parquet("../../data/processed/superstore/data.parquet")
orders = pd.read_parquet("../../data/processed/superstore/orders.parquet")
orders_agg = pd.read_parquet("../../data/processed/superstore/orders_agg.parquet")
dates = pd.read_parquet("../../data/processed/superstore/dates.parquet")
product = pd.read_parquet("../../data/processed/superstore/product.parquet")
product_price_disc = product = pd.read_parquet("../../data/processed/superstore/product_price_disc.parquet")

In [10]:
product_price_disc.columns, orders.columns

(Index(['product_id', 'year', 'month', 'revenue', 'discount', 'profit', 'cost'], dtype='str'),
 Index(['Customer ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
        'Product ID', 'Sales', 'Quantity', 'Discount', 'Profit',
        'days_till_shipped'],
       dtype='str'))

#### Overall discount trend

In [23]:
# determing whether the typical discount offered has changed over time
product_price_disc.groupby(["year"])["discount"].agg(
    ['mean','median']
).round(4).reset_index()

,year,mean,median
0,2014,0.3107,0.2
1,2015,0.3116,0.2
2,2016,0.3152,0.2
3,2017,0.3157,0.2


- The typical discount value increased after 2015. It went from ~0.31 to ~0.31.5
- The discount increased by bit, the revenue growth slowed. Therefore, potentially reducing the revenue generated from each order and contributing to declining MOV

#### Discount by Category

#### Discount Distribution

#### Revenue vs Discount

Discount vs Median Order Value

Correlation